### ISSUE TITLE
Slider/Carousel

### ISSUE THREAD

**Issue by re-becca** [qld-gov-au/qgds-design-roadmap#15]
### Contributing agency/department
Queensland Health - via [@meaghan-s](https://github.com/meaghan-s)

### Overview
Intopia’s accessibility audit of Queensland Health websites in February 2025 identified issues with the use of sliders. Areas of concern were misleading structure, keyboard accessibility, screen reader issues, visual accessibility, no slider indication, lack of feedback and zoom problems. While general best practice states to avoid using sliders (also known as carousels), their use across existing Queensland Health websites led us to interrogate its use case and design.

**Figma file**: https://www.figma.com/design/JRgPvFPibb2mht1l6GyJNZ/Carousel-and-Newsroom-design?node-id=6825-17170&t=rK9fVraxohvD33mz-1 

### Design process
1. Comparative review conducted of in-market sliders (i.e. Apple, Samsung, NSW Govt).

- Key takeaways of common design approaches:  
- clear distinction of current and future points 
- simple and clear stacking and style of chevrons
- partial masking to preview previous and upcoming tiles.

2. Best practice research provided further guidance for accessible sizing, contrast, visibility and keyboard actions (Source: [W3C](https://www.w3.org/WAI/tutorials/carousels/working-example/)).

3. A summary of findings was created, breaking down the above into design requirements (accessibility) and recommendations (good ideas to test).

4. Design exploration was conducted, executing multiple design options within the context of their use case (Queensland Health news cards). These were critiqued with the previous findings in mind.

5. A proposed design was presented to Queensland Health Digital Transformation leadership. This proposed design included rationale for changes to slider buttons, status indicator, screen size stacking and masking, and the introduction of rules for slider build, inputs, and appropriate use case. 

6. With design approval, specifications were made for developers, outlining slider rules and breaking down the design in detail.

7. Progressed to build.

### Key changes

**Slider buttons**: The pre-existing extra-small chevron buttons were enlarged and moved to either side of the tile row. They were also redesigned to have triggering disabled and enabled states at the natural start and end of the slider to aid in providing feedback. The colours and states of the buttons were also designed to look similar to other secondary buttons, which are currently used for iconographic buttons in the design system.

**Status indicator**: Enlarged the extra-small plain circles and added numbers to clearly indicate the number of frames in the slider. This status indicator design also considered differences between screen sizes. A ‘jump point’ design is introduced at the mobile size. The jump point functions like ‘back to start’ or ‘skip to end’. More information on this can be found in the specifications in Figma.

**Row card load**: If the final set of new cards doesn’t entirely fill the container, it keeps cards from the previous frame to fill the gap. For example, if a desktop screen has 3 visible cards per frame, and 7 cards total, the fifth and sixth card accompany the seventh in the final frame. 

**Constraints and use cases**: Introducing a minimum and maximum of cards that are loaded into the slider. When card totals don’t meet the minimum requirement, cards should openly stack. Where possible, we recommend  stacking content instead of using sliders.

### Additional context

For this first implementation, we’ve utilised the disabled button function, based on internal feedback. However, we recognise that user testing is needed to confirm if this interaction is better than an ‘endless loop’ with only active buttons.
